src/components/Header/Header.tsx

```diff
@@ -1,17 +1,67 @@
 import { Metadata, Modal, Player } from '@obr'
 import RefreshIcon from '@icons/refresh.svg?react'
 import SettingsIcon from '@icons/settings.svg?react'
-import { useContext } from 'react'
+import { useContext, useEffect, useState } from 'react'
 import { PlayerContext } from 'context/PlayerContext'
-import { SceneProvider } from 'context/SceneContext'
+import { SceneContext, SceneProvider } from 'context/SceneContext'
+const numberRegex = /\d/
 
 export const Header: React.FC = () => {
   const playerContext = useContext(PlayerContext)
+  const sceneContext = useContext(SceneContext)
+  // headerRound is modeled as a string to allow interim blank states with the html inputs
+  // once a number value is entered, the number value is persisted to the scene context
+  const [headerRound, setRound] = useState<string>('0')
+  const isGM = playerContext?.playerState.role === Player.PlayerRole.GM
+
+  // necessary because on initial render, sceneContext is uninitialized.
+  // this hook updates headerRound to be the persisted value, once context is ready
+  useEffect(() => {
+    if(sceneContext?.roundCount !== undefined) {
+      setRound(`${sceneContext?.roundCount}`)
+    }
+  }, [sceneContext?.roundCount])
+
+  const incrementRound = async (): Promise<void> => {
+    updateRound(`${(+headerRound) + 1}`)
+  }
+
+  const updateRound = async (round: string) : Promise<void> => {
+    if(!round.length) { // allows the input to be blank while the user is interacting with input
+      setRound('')
+      return
+    }
+
+    if(!numberRegex.test(round)) {
+      return
+    }
+
+    // filters out nan or negative numbers
+    if(+round < 0) {
+      round = '0'
+      return
+    }
```

Comment:
can we limit the round to 99? Unless we can find a way to dynamically adjust the input width to the content of the input, but the css changes and limiting to 99 should suffice. I doubt much battles are going to go past 99.